src/main/kotlin/com/john/bookmanagementsystem/feature/book/service/BookService.kt

```diff
@@ -88,11 +104,20 @@ class BookService(
         user.id ?: throw ServiceResponseException("user is not saved", HttpStatus.UNAUTHORIZED)
 
         val borrowLog = borrowLogRepository.findFirstUnreturnedBook(bookId, user.id) ?: throw ServiceResponseException(
-            "Not related book was borrowed",
+            "No related book was borrowed",
             HttpStatus.UNAUTHORIZED
         )
+        val statistics = findBookStatistics(book)
+
+        val returnedDate = LocalDateTime.now()
+        val previousBorrowCount = statistics.borrowCount - 1
+        val averageBorrowTime = (
+                (statistics.averageBorrowTime * previousBorrowCount) +
+                        Duration.between(borrowLog.borrowedDate, returnedDate).seconds
+                ) / statistics.borrowCount
```

Comment:
Add test for borrowing time and for borrowing count.
Tip: Use Zonky instead of embedded